src/libraries/System.Private.CoreLib/src/System/Collections/BitArray.cs

```diff
@@ -0,0 +1,972 @@
+                    }
+                }
+                else if (AdvSimd.Arm64.IsSupported)
+                {
+                    Vector128<byte> ones = Vector128.Create((byte)1);
+                    Vector128<byte> bitMask128 = Vector128.Create(0x80402010_08040201).AsByte();
+
+                    fixed (bool* destination = &boolArray[index])
+                    {
+                        for (; (i + Vector128<byte>.Count * 2u) <= (uint)_bitLength; i += (uint)Vector128<byte>.Count * 2u)
+                        {
+                            int bits = in32Span[(int)(i / (uint)BitsPerInt32)];
+
+                            // Same logic as SSSE3 path, except we do not have Shuffle instruction.
+                            // (TableVectorLookup could be an alternative - dotnet/runtime#1277)
+                            // Instead we use chained ZIP1/2 instructions:
+                            // (A0 is the byte containing LSB, A3 is the byte containing MSB)
+                            // bits                                 - A0 A1 A2 A3
+                            // v1 = Vector128.Create                - A0 A1 A2 A3 A0 A1 A2 A3 A0 A1 A2 A3 A0 A1 A2 A3
+                            // v2 = ZipLow(v1, v1)                  - A0 A0 A1 A1 A2 A2 A3 A3 A0 A0 A1 A1 A2 A2 A3 A3
+                            // v3 = ZipLow(v2, v2)                  - A0 A0 A0 A0 A1 A1 A1 A1 A2 A2 A2 A2 A3 A3 A3 A3
+                            // shuffledLower = ZipLow(v3, v3)       - A0 A0 A0 A0 A0 A0 A0 A0 A1 A1 A1 A1 A1 A1 A1 A1
+                            // shuffledHigher = ZipHigh(v3, v3)     - A2 A2 A2 A2 A2 A2 A2 A2 A3 A3 A3 A3 A3 A3 A3 A3
+
+                            Vector128<byte> vector = Vector128.Create(bits).AsByte();
+                            vector = AdvSimd.Arm64.ZipLow(vector, vector);
+                            vector = AdvSimd.Arm64.ZipLow(vector, vector);
+
+                            Vector128<byte> shuffledLower = AdvSimd.Arm64.ZipLow(vector, vector);
+                            Vector128<byte> extractedLower = AdvSimd.And(shuffledLower, bitMask128);
+                            Vector128<byte> normalizedLower = AdvSimd.Min(extractedLower, ones);
+
+                            Vector128<byte> shuffledHigher = AdvSimd.Arm64.ZipHigh(vector, vector);
+                            Vector128<byte> extractedHigher = AdvSimd.And(shuffledHigher, bitMask128);
+                            Vector128<byte> normalizedHigher = AdvSimd.Min(extractedHigher, ones);
+
+                            AdvSimd.Arm64.StorePair((byte*)destination + i, normalizedLower, normalizedHigher);
+                        }
+                    }
+                }
+
+            Remainder:
+                for (; i < (uint)_bitLength; i++)
+                {
+                    (uint byteIndex, uint extraBits) = Math.DivRem(i, BitsPerByte);
+                    boolArray[(uint)index + i] = (_array[byteIndex] & (1 << (int)extraBits)) != 0;
+                }
```

Comment:
nit: This is a really expensive handler for the remainder and is doing a lot of unnecessary repeat work per byte.

I don't think it needs to be fixed up now, but it would be good to fix it up longer term; so maybe a tracking issue?